assets/src/components/v2/elevator/closures.tsx

```diff
@@ -127,119 +180,126 @@ const CurrentClosures = ({
     }
   });
 
-  // Each index represents a page number and each value represents the number of
-  // rows on the corresponding page index.
-  const [rowCountsPerPage, setRowCountsPerPage] = useState<number[]>([]);
+interface Props extends WrappedComponentProps {
+  id: string;
+  stations_with_closures: StationWithClosures[];
+  station_id: string;
+  upcoming_closure?: UpcomingClosureInfo;
+}
 
-  const numPages = Object.keys(rowCountsPerPage).length;
+const Closures = ({
+  stations_with_closures: stations,
+  station_id: stationId,
+  upcoming_closure: upcomingClosure,
+  updateVisibleData,
+}: Props) => {
+  const numClosuresInStation = stations
+    .filter((s) => s.id === stationId)
+    .flatMap((s) => s.closures).length;
+
+  const listRef = useRef<HTMLDivElement>(null);
+
+  // Number of rows on each page index of the closure list (not counting the
+  // page for an upcoming closure, if there is one).
+  const [pageRowCounts, setPageRowCounts] = useState<number[]>([]);
+
+  // Let the browser lay out the closures into flexbox columns (pages) for us,
+  // then look at their distinct `offsetLeft` values to determine how many pages
+  // we have and how many items are on each.
+  useLayoutEffect(() => {
+    if (!listRef.current) return;
+
+    setPageRowCounts(
+      _.chain(listRef.current.children)
+        .toArray()
+        .map((child) => (child as HTMLElement).offsetLeft)
+        .countBy(_.identity)
+        .values()
+        .value(),
```

Comment:
I'd pull this out into a separate function for readability and especially testability.

Reply:
This is part of my reason for [this Slack message](https://mbta.slack.com/archives/C02SXM166N8/p1740416266196749) -- I've seen unexpected issues with Lodash in the past, and like having unit tests for possible edge cases of chaining together so many functions of a library.

Reply:
Thinking it over, this implementation was overcooked, and in [this commit](https://github.com/mbta/screens/pull/2443/commits/cc29f1170d59cbe045d8654032b5ad772266c0da) I replaced it with something simpler that doesn't use chaining. Do you think this is still worth pulling out a function for? I'm less sure now than I was about the original version.

Reply:
oh nice, that is much simpler! I don't see a need to pull out a function and test it with your update